xmppserver/src/main/java/org/jivesoftware/util/StreamErrorException.java

```diff
@@ -57,6 +57,7 @@ public StreamErrorException(StreamError.Condition condition, String message, Thr
 
     public StreamErrorException(StreamError streamError)
     {
+        super(streamError != null && !streamError.getText().isEmpty() ? streamError.getText() : null);
```

Comment:
Can getText() ever return null?
If it can...

```suggestion
        super(streamError != null && streamError.getText() != null && !streamError.getText().isEmpty() ? streamError.getText() : null);
```

Reply:
Dunno why I'm on this thread, but 
```
 super(streamError != null && streamError.getText() != null && !streamError.getText().isEmpty() ? streamError.getText() : null);
```
Could be expressed a lot neater - to my mind anyway - as 
```
super( ofNullable(streamError).map(StreamError::getText).filter(StringUtils::isNotBlank).orElse(null) );
```

Reply:
@Fishbowler text can be null. I'm not sure if it ever will be null, but it is allowed to be null. To avoid future breakage, I thought it best to code defensively here.

@GregDThomas You're on this thread because I've asked you to review this PR back when it was a single commit. ;)

I'm not a fan of the alternative annotation - it's more complex in my mind. That's probably a lot of personal preference going into this.

Reply:
@guusdk If `getText()` can be null, do we need to add more defensive code, since `getText().isEmpty()` could NPE?